ui/raidboss/data/06-ew/trial/zodiark-ex.ts

```diff
@@ -19,6 +124,307 @@ const triggerSet: TriggerSet<Data> = {
+      netRegex: NetRegexes.startsUsing({ id: '67ED', source: 'Zodiark' }),
+      netRegexDe: NetRegexes.startsUsing({ id: '67ED', source: 'Zodiark' }),
+      netRegexFr: NetRegexes.startsUsing({ id: '67ED', source: 'Zordiarche' }),
+      netRegexJa: NetRegexes.startsUsing({ id: '67ED', source: 'ゾディアーク' }),
+      alertText: (_data, _matches, output) =>
+        output.combo!({
+          dir1: output.northwest!(),
+          dir2: output.southeast!(),
+        }),
+      outputStrings: directionOutputStrings,
+    },
+    {
+      id: 'ZodiarkEx Paradeigma 1',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '67BF', source: 'Zodiark' }),
+      netRegexDe: NetRegexes.startsUsing({ id: '67BF', source: 'Zodiark' }),
+      netRegexFr: NetRegexes.startsUsing({ id: '67BF', source: 'Zordiarche' }),
+      netRegexJa: NetRegexes.startsUsing({ id: '67BF', source: 'ゾディアーク' }),
+      condition: (data) => data.paradeigmaCounter === 0,
+      delaySeconds: 0.1,
+      alertText: (data, _matches, output) => {
+        ++data.paradeigmaCounter;
+        return output.underQuetz!();
+      },
+      outputStrings: {
+        underQuetz: {
+          en: 'Under NW Quetzalcoatl',
+        },
+      },
+    },
+    {
+      id: 'ZodiarkEx Paradeigma 2',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '67BF', source: 'Zodiark' }),
+      netRegexDe: NetRegexes.startsUsing({ id: '67BF', source: 'Zodiark' }),
+      netRegexFr: NetRegexes.startsUsing({ id: '67BF', source: 'Zordiarche' }),
+      netRegexJa: NetRegexes.startsUsing({ id: '67BF', source: 'ゾディアーク' }),
+      condition: (data) => data.paradeigmaCounter === 1,
+      delaySeconds: 0.1,
+      promise: refreshCombatantsPromise,
+      alertText: (data, _matches, output) => {
+        ++data.paradeigmaCounter;
+        const x1 = data.storedQuetzalcoatls[2]?.PosX;
+        const y1 = data.storedQuetzalcoatls[2]?.PosY;
+        const x2 = data.storedQuetzalcoatls[3]?.PosX;
+        const y2 = data.storedQuetzalcoatls[3]?.PosY;
+
+        if (x1 === undefined || x2 === undefined || y1 === undefined || y2 === undefined)
+          return;
+
+        if (x1 < 100 && x2 < 100)
+          return output.west!();
+        if (x1 > 100 && x2 > 100)
+          return output.east!();
+        if (y1 < 100 && y2 < 100)
+          return output.north!();
+        return output.south!();
+      },
+      outputStrings: directionOutputStrings,
+    },
+    {
+      id: 'ZodiarkEx Paradeigma 3',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '67BF', source: 'Zodiark' }),
+      netRegexDe: NetRegexes.startsUsing({ id: '67BF', source: 'Zodiark' }),
+      netRegexFr: NetRegexes.startsUsing({ id: '67BF', source: 'Zordiarche' }),
+      netRegexJa: NetRegexes.startsUsing({ id: '67BF', source: 'ゾディアーク' }),
+      condition: (data) => data.paradeigmaCounter === 2,
+      delaySeconds: 0.1,
+      run: (data) => ++data.paradeigmaCounter,
```

Comment:
Do you think it's worth just having one "Paradeigma Counter" trigger up top that handles this in all places, vs conditionally incrementing everywhere?

Reply:
Wouldn't that run into the same race condition as the reason I added the 0.1 second delay?

Reply:
If anything, I think it would address it, if I'm understanding this correctly.  I think there are two ways to remove the `delaySeconds: 0.1`.  One is to reverse order your paradeigmas in the file so that they can never chain into each other by having 0 fire, increment, and then 1 fire.  The other is to have a single counter incrementing trigger at the top above everything (so that you only increment once), and then have all of the following triggers just check (but not increment).  Since they would all be checking different values, only one would fire.